### PR TITLE
Use new type assert functions in `Repository`

### PR DESCRIPTION
**Pull request by trim21**

### CODE REVIEW

**Comment by trim21** [comment]
@EnricoMi

File: github/Repository.py [hunk 1]
```diff
@@ -2324,9 +2256,7 @@ def get_views_traffic(self, per: Opt[str] = NotSet) -> None | dict[str, int | li
         :calls: `GET /repos/{owner}/{repo}/traffic/views <https://docs.github.com/en/rest/reference/repos#traffic>`_
         :param per: string, must be one of day or week, day by default
         """
-        assert is_undefined(per) or (
-            isinstance(per, str) and (per == "day" or per == "week")
-        ), "per must be day or week, day by default"
+        assert is_undefined(per) or per in ("day", "week"), "per must be day or week, day by default"
```
Comment (EnricoMi) [inline github/Repository.py:2259]:
```suggestion
        assert per in ["day", "week", NotSet], "per must be day or week, day by default"
```

File: github/Repository.py [hunk 2]
```diff
@@ -2343,9 +2273,7 @@ def get_clones_traffic(self, per: Opt[str] = NotSet) -> dict[str, int | list[Clo
         :param per: string, must be one of day or week, day by default
         :rtype: None or list of :class:`github.Clones.Clones`
         """
-        assert is_undefined(per) or (
-            isinstance(per, str) and (per == "day" or per == "week")
-        ), "per must be day or week, day by default"
+        assert is_undefined(per) or per in ("day", "week"), "per must be day or week, day by default"
```
Comment (EnricoMi) [inline github/Repository.py:2276]:
```suggestion
        assert per in ["day", "week", NotSet], "per must be day or week, day by default"
```

File: github/Repository.py [hunk 3]
```diff
@@ -2821,23 +2745,16 @@ def get_issues(
         :rtype: :class:`PaginatedList` of :class:`github.Issue.Issue`
         """
         assert (
-            is_undefined(milestone)
-            or milestone == "*"
-            or milestone == "none"
-            or isinstance(milestone, github.Milestone.Milestone)
+            is_undefined(milestone) or milestone in ("*", "none") or isinstance(milestone, github.Milestone.Milestone)
```
Comment (EnricoMi) [inline github/Repository.py:2748]:
```suggestion
            milestone in ["*", "none", NotSet] or isinstance(milestone, github.Milestone.Milestone)
```

**Review by EnricoMi — CHANGES_REQUESTED** [review]
Looks good, some minor adjustments.

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Pull request merged**